### PR TITLE
submit hibernate cfg

### PR DESCRIPTION
**Pull request by hangzhang925**
re-submite hibernate cfg and examples.
Have fixed all issues from comments.

### CODE REVIEW

File: wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java [hunk 1]
```diff
@@ -13,9 +13,13 @@
  */
 package wherehows.dao;
 
+import javax.persistence.EntityManagerFactory;
+
+
 public class DaoFactory {
 
   private static DatasetsDao datasetsDao;
+  private static EntityManagerFactory entityManagerFactory;
```
Comment (mars-lan) [inline wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java]: private final

Comment (hangzhang925) [inline wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java]: done

File: wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java [hunk 2]
```diff
@@ -24,4 +28,11 @@ public DatasetsDao getDatasetsDao() {
     return datasetsDao;
   }
```
Comment (mars-lan) [inline wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java]: Sorry one more thing, add getDatasetClassificationDao() function here and use it in GobblinTrackingAuditProcessor.java

File: wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java [hunk 3]
```diff
@@ -24,4 +28,11 @@ public DatasetsDao getDatasetsDao() {
...
 
+  public DaoFactory(EntityManagerFactory entityManagerFactory) {
+    this.entityManagerFactory = entityManagerFactory;
+  }
+
+  public EntityManagerFactory getEntityManagerFactory() {
```
Comment (mars-lan) [inline wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java]: Then you can remove the getter

File: wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java (hunk 2 above)
Comment (hangzhang925) [inline wherehows-dao/src/main/java/wherehows/dao/DaoFactory.java]: OK, understand the way how i should use the daoFactory now.

File: wherehows-dao/src/main/java/wherehows/models/DatasetClassification.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+import javax.persistence.Id;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+
+@Data
+@Entity
+@Table(name = "dataset_classification")
+@NoArgsConstructor
```
Comment (alyiwang) [inline wherehows-dao/src/main/java/wherehows/models/DatasetClassification.java]: One more thing, can probably do it in next PR, @Data actually has @RequiredArgsConstructor, and in this case is equal to @NoArgsConstructor here.

Comment (hangzhang925) [inline wherehows-dao/src/main/java/wherehows/models/DatasetClassification.java]: OK, thanks for the info, will do it in next PR.

**Pull request merged**